server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/Session.scala

```diff
@@ -31,30 +35,58 @@ import org.apache.james.jmap.model.State.State
 case class IsPersonal(value: Boolean)
 case class IsReadOnly(value: Boolean)
 
-object Account {
-  def apply(name: Username,
-            isPersonal: IsPersonal,
-            isReadOnly: IsReadOnly,
-            accountCapabilities: Set[_ <: Capability]): Account = {
+object AccountId {
+  def from(username: Username): Either[IllegalArgumentException, AccountId] = {
+    val sha1String = Hashing.sha1()
+      .hashString(username.asString(), StandardCharsets.UTF_8)
+      .toString
+    val refinedId: Either[String, Id] = refineV(sha1String)
```

Comment:
is `Id` defined as a sha1? And BTW, why sha1 as it's known to be weak?

Reply:
we are considering that is a temporal solution, JMAP spec requires an `AccountId` associating with a `username`. We don't have `AccountId` at hand, so, to make it being some how mapping with a `username` on the fly, I hashed the `username as string`.

> And BTW, why sha1 as it's known to be weak?

yeah it's weak, but fast, as we're already exposing the source `username`, why we need to secure the hash?

Reply:
> yeah it's weak, but fast, as we're already exposing the source username, why we need to secure the hash?

preventing collision in future usages

Reply:
okay, is `SHA-256` good enough?

Reply:
>  why we need to secure the hash?

Because `@` is not supported in AccountId:

```
Where Id is given as a data type, it means a String of at least 1 and a maximum of 255 octets in size, and it MUST only contain characters from the “URL and Filename Safe” base64 alphabet, as defined in Section 5 of [@!RFC4648], excluding the pad character (=). This means the allowed characters are the ASCII alphanumeric characters (A-Za-z0-9), hyphen (-), and underscore (_).
```

Maybe base64 encoding would allow a bijection and thus be safer?

Reply:
@chibenwa I don't see how the question and you answer matching each other

Reply:
BTW base64 strings might contain `=` at the end which is a forbidden character. And the length of base64 string will probably larger than the source, with sources having length closing to 255, base64 strings will break `ID` validation.

Reply:
I'll take `SHA-256`